addons/product/models/product_product.py

```diff
@@ -134,7 +134,7 @@ def _compute_write_date(self):
         works as intended :-)
         """
         for record in self:
-            record.write_date = max(record.write_date, record.product_tmpl_id.write_date)
+            record.write_date = max(record.write_date or fields.Datetime.now(), record.product_tmpl_id.write_date)
```

Comment:
```suggestion
            record.write_date = max(record.write_date or self.env.cr.now(), record.product_tmpl_id.write_date)
```

Reply:
Hello @Feyensv 

Suggested changes are done please review.

Thank You :innocent: